tests/php/test_class.jetpack_photon.php

```diff
@@ -11,13 +11,30 @@ public function setUp() {
 		// Preserving global variables
 		global $content_width;
 		$this->_globals['content_width'] = $content_width;
+
+		// Setup the Photon filters.
+		// WP_UnitTestCase resets the action/filter state after
+		// every test:
+		// https://core.trac.wordpress.org/browser/trunk/tests/phpunit/includes/testcase.php?rev=43005#L273
+		// So we need to set these Photon filters for each test.
+		// see ::tearDown() ...
+		Jetpack_Photon::instance();
 	}
 
 	public function tearDown() {
 		// Restoring global variables
 		global $content_width;
 		$content_width = $this->_globals['content_width'];
 
+		// ... see ::setUp()
+		// Unfortunately Jetpack_Photon::instance() won't run Jetpack_Photon->setup()
+		// each time Jetpack_Photon::instance() is called, since it's gated by a
+		// static variable.
+		// l337 h4X0Ring required:
+		$instance = new ReflectionProperty( 'Jetpack_Photon', '__instance' );
+		$instance->setAccessible( true );
```

Comment:
This makes our PHP 5.2 tests trip:

```
Fatal error: Call to undefined method ReflectionProperty::setAccessible() in wp-content/plugins/jetpack/tests/php/test_class.jetpack_photon.php on line 35
```

Reply:
06e261d29 may "fix" it.

Reply:
I am afraid the tests are still failing :( Travis won't let us merge that one.

Reply:
OK - I'll look into it more later. Thanks for the ping!

Reply:
Looks like it's working as of ea220aa376bc9eb2209aebce6193b50876b7122f.